src/core/dbus-manager.c

```diff
@@ -1492,6 +1492,12 @@ static int method_reload(sd_bus_message *message, void *userdata, sd_bus_error *
         if (r == 0)
                 return 1; /* No authorization for now, but the async polkit stuff will call us again when it has it */
 
+        /* Check the rate limit after the authorization succeeds, to avoid denial-of-service issues. */
+        if (!ratelimit_below(&m->reload_ratelimit))
+                return sd_bus_error_setf(error,
+                                         SD_BUS_ERROR_LIMITS_EXCEEDED,
+                                         "Reload() request rejected due to too frequent reloading.");
```

Comment:
I think we should not only return an error here but *also* log that we hit the ratelimit, and do so indicating the client which caused the limit to be hit.

After all the ratelimit is a global concept, hence we should surface it not only locally for the method call but globally in system logs, too